### DISCUSSION TITLE
Sample Queries

### DISCUSSION THREAD

**Discussion by wallaceturner** [ravendb/ravendb#18522] [General]
To the 'Sample Queries' can I ask that you add an example that is equaivalent to a 'string contains' ?
e.g.

'return results that contain 'widget' anywhere in the `Notes` field, case-insensitive'
```
from Orders where 
search(Notes, "*widget*")
```

(please advise also if this IS the correct query)

**Comment by ayende** [comment]
This is the query you want:

```
from Orders where 
search(Notes, "widget")
```

Note that in this case, you relying on RavenDB's full text search to prevent a "contains" operation. This is broken on word boundary and then you can query for the actual term you are looking for.

A string contains is a lot more expensive, however, since we need to scan all the items in the index. 

If you are looking at a partial search, however, you can also use the ngram analyzer as well.

**Comment by wallaceturner** [comment]
that search does not produce the desired result (in this case)
for instance suppose you were querying all email addresses containing 'hotmail'

```
from Orders where 
search(Email, "*hotmail*")
```

works - i.e. it will find bob@hotmail,com however without the asterix no results are returned:

```
from Orders where 
search(Email, "hotmail")
```

**Reply by ayende** [reply]
For emails, we are keeping the email as is, that is why you'll see this behavior.
There are two options that you can take. using ngrams, see: https://ravendb.net/docs/article-page/6.0/csharp/indexes/using-analyzers#analyzers-that-tokenize-according-to-the-defined-number-of-characters

Or use:

```
from Orders where 
lucene(Email, "*hotmail*")
```

Note that this is an expensive operation.

**Reply by wallaceturner** [reply]
duly noted about the expense.

`search(Email, "hotmail")` appears to work too - does this simply use lucene under the hood? If it doesnt work properly under what scenario won't it work?

as an aside the reason for asking for this in the 'Sample Queries' as its quite a common operation

**Reply by ayende** [reply]
Note that our sample queries are running on the sample data, we don't _have_ an Email field there.
That said, the problem with sample queries ( I assume you mean in the Studio) is that we don't have sufficient space to put everything there.
